### DISCUSSION TITLE
The plugin is executed several times on the server in case there are broken images

### DISCUSSION THREAD

**Discussion by milelazar** [nuxt/nuxt#19889] [Questions] [answered]
Not sure if this is a bug or expected behavior. 

The client has the ability to edit content and from time to time it may happen that some images are no longer available.
But in case there are broken images, plugin is executed multiple times on the server side. This happens on every refresh or first page load.

Problem is even bigger if you have some api fetch calls in the plugin. Each fetch is executed multple times on the server. Depending on how many broken images there are.

Here is a reproduction:
https://stackblitz.com/edit/github-mb4zkh?file=app.vue

Is there a way to resolve this? "Tell the client to fix the broken images" is not the answer I expect :)

**Comment by danielroe** [comment, marked as answer]
The reason this is occuring for you is that:

1. you do not have a `pages/` directory so there is no routing. All requests are handled by app.vue if they don't match a static asset.
2. The request for the image (which doesn't exist in your Nuxt project) is therefore hitting the Nuxt server.
3. The plugin is therefore running for each image, because the server is rendering HTML for that request.

One way to resolve this would be to write a server middleware (e.g. `~/server/middleware/images.ts`) that early returns with a 404 if the request path ends in an image file extension.

Another would be to enable the router by moving your `app.vue` to `pages/index.vue`.

**Reply by milelazar** [reply]
Thanks for the solutions.

Enabling router didn't help (I have updated the stackblitz example). Every time you end up on homepage where broken images are (either on first load or using navigation), the plugin is executed multiple times.

But the server middleware solution works so I went with that.
